src/EStimLibrary/Core/ResourceManager.cs

```diff
@@ -8,13 +8,15 @@ public class ResourceManager<ResourceType>
 
     public readonly int MaxNumResources;
 
-    public int NumTotalResources => this.IdPool.NumUsedIds;
+    public int NumTotalResources => this.Resources.Count;
```

Comment:
Was the previous implementation of this property failing? If so, something is broken, because in this case, IdPool.NumUsedIds and Resources.Count should always be the same